sdk/core/core/src/az_log.c

```diff
@@ -14,8 +14,8 @@
 
 #include <_az_cfg.h>
 
-static az_log_classification const* _az_log_classifications = NULL;
-static az_log_message_fn _az_log_message_callback = NULL;
+static az_log_classification const* volatile _az_log_classifications = NULL;
+static az_log_message_fn volatile _az_log_message_callback = NULL;
```

Comment:
Do we have a test that highlights why these need to be volatile?

Reply:
No, you can't create a test for timing

Reply:
OK. Can you please explain why these need to be volatile?

Reply:
One thread sets the value while another thread reads the value. The use of volatile ensures that this works correctly in that the compiler won't cache the value in a register or temporary variable.

Reply:
Your question made me think about the code and I modified it and sent revised code to @antkmsft to test and check in.